src/commonTest/kotlin/fr/acinq/lightning/channel/TestsHelper.kt

```diff
@@ -64,12 +60,84 @@ internal inline fun <reified T : ChannelAction> List<ChannelAction>.find() = fin
+        state
+            .run { ctx.process(cmd) }
+            .let { (newState, actions) ->
+                checkSerialization(actions)
+                JsonSerializers.toJsonString(newState)
+                LNChannel(ctx, newState) to actions
+            }
+
+    // we check that serialization works by checking that deserialize(serialize(state)) == state
+    private fun checkSerialization(state: PersistedChannelState) {
+
+        // We never persist remote channel data.
+        fun removeChannelData(state: PersistedChannelState): PersistedChannelState = when (state) {
+            is LegacyWaitForFundingConfirmed -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is LegacyWaitForFundingLocked -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is WaitForFundingConfirmed -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is WaitForChannelReady -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is Normal -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is ShuttingDown -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is Negotiating -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is Closing -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is WaitForRemotePublishFutureCommitment -> state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty))
+            is Closed -> state.copy(state = state.state.copy(commitments = state.commitments.copy(remoteChannelData = EncryptedChannelData.empty)))
+        }
+
+        // We never persist a funding RBF attempt.
+        fun removeRbfAttempt(state: PersistedChannelState): PersistedChannelState = when (state) {
+            is WaitForFundingConfirmed -> state.copy(rbfStatus = WaitForFundingConfirmed.Companion.RbfStatus.None)
+            else -> state
+        }
+
+        val serialized = Serialization.serialize(state)
+        val deserialized = Serialization.deserialize(serialized)
+
+        assertEquals(removeChannelData(removeRbfAttempt(state)), deserialized, "serialization error")
+    }
+
+    private fun checkSerialization(actions: List<ChannelAction>) {
+        // we check that serialization works everytime we're supposed to persist channel data
+        actions.filterIsInstance<ChannelAction.Storage.StoreState>().forEach { checkSerialization(it.data) }
+    }
+
+}
+
+// same as process but with added assumptions on exit state
+fun <T : ChannelState> LNChannel<T>.processSameState(event: ChannelCommand): Pair<LNChannel<T>, List<ChannelAction>> {
```

Comment:
nit: why isn't that function inside `LNChannel` (right below the `process` one)?

Reply:
For no good reason, changed in c5365f4.